### PR TITLE
[phenyl-mongodb] convert find operation from DocumentPath to DotNotation

### LINKED ISSUE #44: Bridge the gap between DocumentPath and MongoDB ( array index expression )
Implement mongodb-client to bridge the following difference.

DocumentPath
```js
{ foo.bar[0].baz: 'bazz' }
```

MongoDB
```js
{ foo.bar.0.baz: 'bazz' }
```

[readme](https://github.com/phenyl-js/phenyl/tree/master/modules/power-filter#documentpath)

### PR DESCRIPTION
**Pull request by fitzr**
close #44

### CODE REVIEW

File: modules/phenyl-mongodb/src/mongodb-client.js [hunk 1]
```diff
@@ -17,14 +17,33 @@ import type {
...
+
+function convertDocumentPathToDotNotation(simpleFindOperation: SimpleFindOperation): SimpleFindOperation {
+  return Object.keys(simpleFindOperation).reduce((operation, srcKey) => {
+    const dstKey = srcKey.replace(/\[(\d+)]/g, (m, p) => `.${p}`)
+    operation[dstKey] = simpleFindOperation[srcKey]
+    return operation
+  }, {})
+}
+
+const WHERE_FILTERS : Array<(SimpleFindOperation) => SimpleFindOperation> = [
```
Comment (shinout) [inline modules/phenyl-mongodb/src/mongodb-client.js]: Maybe composing functions is better, but this implementation is ok.

File: modules/phenyl-mongodb/test/mongodb-client.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    const actual = filterWhere(input)
+    assert.deepEqual(actual, expected)
+  })
+
+  it ('converts document path to dot notation', () => {
+    const input: AndFindOperation = {
+      $and: [
+        { 'values[0]': 'fizz' },
+        { 'values[1].test': 'buzz' },
+        { 'values[1234].test': 'fizzBuzz' },
```
Comment (shinout) [inline modules/phenyl-mongodb/test/mongodb-client.js]: Why don't you use query operators in test cases?

```js
{ 'values[1234].test': { $regex: /fizzBuzz/ } }
```

File: modules/phenyl-mongodb/test/mongodb-client.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
+// @flow
+
+import { it, describe } from 'kocha'
+import assert from 'power-assert'
+import { filterWhere } from '../src/mongodb-client'
```
Comment (shinout) [inline modules/phenyl-mongodb/test/mongodb-client.js]: We prefer adding `.js` extension in importing. `../src/mongodb-client.js`

File: modules/phenyl-mongodb/test/mongodb-client.js (hunk 2 above)
Comment (fitzr) [reply modules/phenyl-mongodb/test/mongodb-client.js]: I added test cases.

File: modules/phenyl-mongodb/src/mongodb-client.js (hunk 1 above)
Comment (fitzr) [reply modules/phenyl-mongodb/src/mongodb-client.js]: 96368e5 make sense?

**Comment by shinout** [comment]
LGTM!

**Pull request merged**